src/Planners/src/UnicycleTrajectoryPlanner.cpp

```diff
@@ -0,0 +1,841 @@
+    ok = ok && loadParam("rightZMPDelta", rightZMPDelta);
+    ok = ok && loadParamWithFallback("lastStepDCMOffset", lastStepDCMOffset, 0.5);
+    ok = ok && loadParam("leftContactFrameName", leftContactFrameName);
+    ok = ok && loadParam("rightContactFrameName", rightContactFrameName);
+
+    // try to configure the planner
+    auto unicyclePlanner = m_pImpl->generator.unicyclePlanner();
+
+    ok = ok
+         && unicyclePlanner->setDesiredPersonDistance(m_pImpl->parameters.referencePointDistance[0],
+                                                      m_pImpl->parameters.referencePointDistance[1]);
+    ok = ok && unicyclePlanner->setPersonFollowingControllerGain(unicycleGain);
+    ok = ok && unicyclePlanner->setSlowWhenTurnGain(slowWhenTurningGain);
+    ok = ok && unicyclePlanner->setSlowWhenBackwardFactor(slowWhenBackwardFactor);
+    ok = ok && unicyclePlanner->setSlowWhenSidewaysFactor(slowWhenBackwardFactor);
+    ok = ok && unicyclePlanner->setMaxStepLength(maxStepLength, maxLengthBackwardFactor);
+    ok = ok && unicyclePlanner->setMaximumIntegratorStepSize(m_pImpl->parameters.dt);
+    ok = ok && unicyclePlanner->setWidthSetting(minWidth, m_pImpl->parameters.nominalWidth);
+    ok = ok && unicyclePlanner->setMaxAngleVariation(maxAngleVariation);
+    ok = ok && unicyclePlanner->setMinimumAngleForNewSteps(minAngleVariation);
+    ok = ok && unicyclePlanner->setCostWeights(positionWeight, timeWeight);
+    ok = ok && unicyclePlanner->setStepTimings(minStepDuration, maxStepDuration, nominalDuration);
+    ok = ok && unicyclePlanner->setPlannerPeriod(m_pImpl->parameters.dt);
+    ok = ok && unicyclePlanner->setMinimumStepLength(minStepLength);
+    ok = ok
+         && unicyclePlanner->setSaturationsConservativeFactors(saturationFactors(0),
+                                                               saturationFactors(1));
+    unicyclePlanner->setLeftFootYawOffsetInRadians(m_pImpl->parameters.leftYawDeltaInRad);
+    unicyclePlanner->setRightFootYawOffsetInRadians(m_pImpl->parameters.rightYawDeltaInRad);
+    unicyclePlanner->addTerminalStep(terminalStep);
+    unicyclePlanner->startWithLeft(startWithLeft);
+    unicyclePlanner->resetStartingFootIfStill(startWithSameFoot);
+
+    UnicycleController unicycleController;
+    ok = ok && setUnicycleControllerFromString(unicycleControllerAsString, unicycleController);
+    ok = ok && unicyclePlanner->setUnicycleController(unicycleController);
+
+    ok = ok && m_pImpl->generator.setSwitchOverSwingRatio(switchOverSwingRatio);
+    ok = ok && m_pImpl->generator.setTerminalHalfSwitchTime(lastStepSwitchTime);
+    ok = ok && m_pImpl->generator.setPauseConditions(maxStepDuration, nominalDuration);
+    ok = ok && m_pImpl->generator.setMergePointRatio(mergePointRatios[0], mergePointRatios[1]);
+
+    m_pImpl->generator.setPauseActive(isPauseActive);
+
+    m_pImpl->parameters.leftContactFrameIndex
+        = m_pImpl->kinDyn.model().getFrameIndex(leftContactFrameName);
```

Comment:
Ok you need the model for getting the index of the frames. Perhaps we may discuss this also with @S-Dafarra

Reply:
Indeed, the frame index, such as `m_pImpl->parameters.leftContactFrameIndex` is then used only for building the `contactList`. It is not used by the [unicycle-footsteps-planner](https://github.com/robotology/unicycle-footstep-planner).

I agree that it could be an overkill to load and store the robot model just to extract the frame index.

I do not have yet come up with a clean way to get rid of it.

Reply:
Fixed with https://github.com/ami-iit/bipedal-locomotion-framework/commit/1537612b867967e61d33818362281f7dc9bdc40f. I have added a member function called `SetRobotContactFrames`